This needs a LOT more work before it is really usable.

--- origsrc/native/Makefile	2009-08-04 09:40:08.000000000 -0500
+++ src/native/Makefile	2009-11-29 23:07:47.157273700 -0600
@@ -20,7 +20,8 @@
 # protection simply by defining HAVE_PROTECTION.  This has been enabled
 # only for those platforms on which it has been tested successfully.
 
-OS=$(shell uname | sed -e 's/\(CYGWIN\|MINGW32\).*/win32/g' \
+OS=$(shell uname | sed -e 's/MINGW32.*/win32/g' \
+                        -e 's/CYGWIN.*/cygwin/g' \
                         -e 's/SunOS.*/solaris/g' \
                         -e 's/FreeBSD.*/freebsd/g' \
                         -e 's/OpenBSD.*/openbsd/g' \
@@ -125,6 +126,13 @@ endif
 endif
 endif
 
+ifeq ($(OS),cygwin)
+ARCH=$(shell uname -m | sed 's/i.86/i386/g')
+LDFLAGS=-o $@ -shared 
+LIBPFX=cyg
+LIBSFX=.dll
+endif
+
 ifeq ($(OS),linux)
 ARCH=$(shell uname -m | sed 's/i.86/i386/g')
 PCFLAGS+=-fPIC
--- origsrc/src/com/sun/jna/NativeLibrary.java	2009-09-21 13:04:48.000000000 -0500
+++ src/src/com/sun/jna/NativeLibrary.java	2009-11-29 23:56:42.029588200 -0600
@@ -225,6 +225,13 @@ public class NativeLibrary {
         if (Platform.isLinux() && "c".equals(libraryName)) {
             libraryName = null;
         }
+        if (Platform.isCygwin()) {
+            if ("c".equals(libraryName) || "dl".equals(libraryName) ||
+                "m".equals(libraryName) || "pthread".equals(libraryName) ||
+                "resolv".equals(libraryName) || "rt".equals(libraryName) ||
+                "util".equals(libraryName))
+                libraryName = "/usr/bin/cygwin1.dll";
+        }
         synchronized (libraries) {
             WeakReference ref = (WeakReference)libraries.get(libraryName + options);
             NativeLibrary library = ref != null ? (NativeLibrary)ref.get() : null;
--- origsrc/src/com/sun/jna/Platform.java	2009-06-26 17:54:58.000000000 -0500
+++ src/src/com/sun/jna/Platform.java	2009-11-29 23:37:19.900361800 -0600
@@ -19,6 +19,7 @@ public final class Platform {
     public static final int FREEBSD = 4;
     public static final int OPENBSD = 5;
     public static final int WINDOWSCE = 6;
+    public static final int CYGWIN = 7;
 
     private static final int osType;
     
@@ -30,6 +31,9 @@ public final class Platform {
         else if (osName.startsWith("Mac") || osName.startsWith("Darwin")) {
             osType = MAC;
         }
+        else if (osName.startsWith("CYGWIN")) {
+            osType = CYGWIN;
+        } 
         else if (osName.startsWith("Windows CE")) {
             osType = WINDOWSCE;
         }
@@ -59,6 +63,9 @@ public final class Platform {
     public static final boolean isLinux() {
         return osType == LINUX;
     }
+    public static final boolean isCygwin() {
+        return osType == CYGWIN;
+    }
     public static final boolean isWindowsCE() {
         return osType == WINDOWSCE;
     }
